### PR TITLE
Confirm before deleting a twiddle

### PR DESCRIPTION
**Pull request by Gaurav0**
Fixes issue #87

### CODE REVIEW

**Comment by rwjblue** [comment]
Can you add a test for this?

A unit test will likely be easiest (you could mock `confirm` and pass in an object with a `destroyRecord` method then assert it is called when `confirm` returns true and not called when `confirm` returns false), but an acceptance test would be better.  Acceptance test would be harder, but still should be doable (similar to the changes made in #85).

**Comment by Gaurav0** [comment]
@rwjblue Unit test added.

File: tests/unit/gist/controller-test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+});
+
+// Replace this with your real tests.
+test('it exists', function(assert) {
+  var controller = this.subject();
+  assert.ok(controller);
+});
+
+test('deleting a gist requires confirmation', function(assert) {
+  let cacheConfirm = window.confirm;
```
Comment (rwjblue) [inline tests/unit/gist/controller-test.js]: We should grab this in the `beforeEach` and reset it in the `afterEach`.  The main reason is that if an error is thrown during the test `window.confirm` will not be reset (and will likely cause other tests to fail in the future).

Something like:

```javascript
moduleFor('controller:gist', {
  needs: ['service:ember-cli'],
  beforeEach() {
    this._originalConfirm = window.confirm;
  },

  afterEach() {
    window.confirm = this._originalConfirm;
  }
});
```

File: tests/unit/gist/controller-test.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import Ember from "ember";
+import { moduleFor, test } from 'ember-qunit';
+
+moduleFor('controller:gist', {
+  // Specify the other units that are required for this test.
+  // needs: ['controller:foo']
```
Comment (rwjblue) [inline tests/unit/gist/controller-test.js]: Can you delete this comment?

File: tests/unit/gist/controller-test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  needs: ['service:ember-cli']
+});
+
+// Replace this with your real tests.
```
Comment (rwjblue) [inline tests/unit/gist/controller-test.js]: This test doesn't add value now that you added a real one below, can you delete it?

**Comment by rwjblue** [comment]
@Gaurav0 - Awesome, thank you!  I left a few small comments, but this is looking good!

**Comment by Gaurav0** [comment]
@rwjblue Comments addressed.

**Comment by rwjblue** [comment]
:+1: - Thank you!

**Comment by abidmix** [comment]
Thanks for fixing this prompty.

**Pull request merged**